sql/wsrep_applier.cc

```diff
@@ -124,23 +147,26 @@ static void wsrep_store_error_da(const THD* const thd,
   char* slider= dst.data();
   const char* const buf_end= slider + max_len - 1; // -1: leave space for \0
 
-  for (cond= it++; cond && slider < buf_end; cond= it++)
+  if (cond)
   {
-    uint const err_code= cond->get_sql_errno();
-    const char* const err_str= cond->get_message_text();
-
-    if (include_msg)
-    {
-      slider+= snprintf(slider, buf_end - slider, " %s, Error_code: %d;",
-                        err_str, err_code);
-    }
-    else
+    for (; cond && slider < buf_end; cond= it++)
     {
-      slider+= snprintf(slider, buf_end - slider, " Error_code: %d;",
-                        err_code);
+      uint const err_code= cond->get_sql_errno();
+      const char* const err_str= cond->get_message_text();
+      slider+= wsrep_print_error(slider, buf_end-slider,
+                                 include_msg, err_code, err_str);
     }
   }
-
+  else if (da->is_error())
```

Comment:
This, as far as I can see, changes the voting protocol and during rolling upgrade the nodes may vote "inconsistently". However without this change "consistency" means randomness, hence voting non-randomply won't alter voting results in the worse direction. So I guess we can give it a go without bumping application protocol.